Sources/TestHelper/CoreLocation.swift

```diff
@@ -134,14 +134,29 @@ extension Array where Element == CLLocation {
     }
     
     // Shifts the [CLLocation]’s first location to the given timestamp and offsets the remaining locations by one second after the prior.
-    public func shifted(to timestamp: Date) -> [CLLocation] {
-        return enumerated().map { CLLocation(coordinate: $0.element.coordinate,
-                                             altitude: $0.element.altitude,
-                                             horizontalAccuracy: $0.element.horizontalAccuracy,
-                                             verticalAccuracy: $0.element.verticalAccuracy,
-                                             course: $0.element.course,
-                                             speed: $0.element.speed,
-                                             timestamp: timestamp + $0.offset) }
+    public func shifted(to firstTimestamp: Date) -> [CLLocation] {
+        var previousLocation: CLLocation?
+        var shiftedLocations: [CLLocation] = []
+        for location in self {
+            let timestamp: Date
+            if let previousLocation = previousLocation {
+                timestamp = previousLocation.timestamp + previousLocation.coordinate.distance(to: location.coordinate) / previousLocation.speed
```

Comment:
If I’m reading this correctly, we were previously adding 1&nbsp;second to each location’s timestamp, assuming the frequency that Core Location normally runs on. This instead changes the offset to be based on interpolation. This change could be problematic, because it assumes that `CLLocation.speed` can help predict the next location’s timestamp, only to be used for informational purposes (like displaying your current speed next to a speed limit). But [`CLLocation.speed`](https://developer.apple.com/documentation/corelocation/cllocation/1423798-speed) is only the instantaneous speed reading from GPS. I don’t see how it would be any more reliable than the 1&nbsp;second assumption, unless we happened to hard-code timestamps and speeds in the fixtures that make that assumption true.

Reply:
Thanks for the feedback. I've got in a loop trying different approaches in the hope of fixing failing tests and went too far. So I will revisit this code.

Reply:
I removed these changes for now.